### PR TITLE
Percent wetted and percent shaded

### PR DESCRIPTION
**Pull request by geanderson-062**

### CODE REVIEW

File: irrigation_design_system_backend/core/domain/entity/PWAEntity.py
Comment (ravellys) [inline irrigation_design_system_backend/core/domain/entity/PWAEntity.py:1]: change the file name to the project pattern
percent_wetted_area_entity

File: irrigation_design_system_backend/core/domain/entity/PWAEntity.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+from _decimal import Decimal
+from dataclasses import dataclass
+
+
+@dataclass
+class IBTEntity:
```
Comment (ravellys) [inline irrigation_design_system_backend/core/domain/entity/PWAEntity.py:6]: improve the name of the entities, to be more clear.

File: irrigation_design_system_backend/infrastructure/api/v1/responses/percent_wetted_area.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+from pydantic import BaseModel
+
+class IBTResponse(BaseModel):
```
Comment (ravellys) [inline irrigation_design_system_backend/infrastructure/api/v1/responses/percent_wetted_area.py:3]: fix the name of response, follow the pattern

File: irrigation_design_system_backend/apps/percent_wetted_area/Readme.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+## Methods
+
+### `calculate_irrigation_by_tree(ibt_entity: IBTEntity) -> Decimal`
+
+Calculates the irrigation required for a single tree based on the given parameters.
+
+**Parameters:**
+
+- `ibt_entity`: An instance of the `IBTEntity` class containing the following attributes:
```
Comment (ravellys) [inline irrigation_design_system_backend/apps/percent_wetted_area/Readme.md:13]: fix these names

File: irrigation_design_system_backend/apps/percent_wetted_area/Readme.md
Comment (ravellys) [inline irrigation_design_system_backend/apps/percent_wetted_area/Readme.md:1]: 💯

File: irrigation_design_system_backend/apps/percent_shaded_area/Readme.md
Comment (ravellys) [inline irrigation_design_system_backend/apps/percent_shaded_area/Readme.md:1]: 💯

Comment (ravellys) [inline irrigation_design_system_backend/apps/percent_shaded_area/Readme.md:1]: suggestion: create a flowchart with the input and output of the process. Similar with the thesis literature

**Review by ravellys — CHANGES_REQUESTED** [review]
Create a unit test to the services, similar to the evapotranspiration

File: irrigation_design_system_backend/tests/unit/apps/percent_wetted_area/test_reference_percent_wetted_area_service.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        input_entity = IrrigationTreeEntity(
+            drippers_number=Decimal(4),
+            space_between_lines=Decimal(2),
+            space_between_plants=Decimal(1.5),
+            z=Decimal(0.6),
+            q=Decimal(0.02),
+            hydraulic_conductivity_of_saturated_soil=Decimal(10)
+        )
+        dw_expected = PercentWettedAreaService._dw_calculate(input_entity)
+        expected_result = Decimal(
```
Comment (ravellys) [inline irrigation_design_system_backend/tests/unit/apps/percent_wetted_area/test_reference_percent_wetted_area_service.py:25]: same here

File: irrigation_design_system_backend/infrastructure/api/v1/responses/percent_wetted_area.py
Comment (ravellys) [inline irrigation_design_system_backend/infrastructure/api/v1/responses/percent_wetted_area.py:1]: 💯

**Review by ravellys — COMMENTED** [review]
💯

File: irrigation_design_system_backend/tests/unit/apps/percent_shaded_area/test_reference_percent_shaded_area_service.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    error = Decimal("1e-2")
+
+    def test_calculate_by_plant_canopy_projection(self):
+        # Given
+        input_entity = PlantCanopyProjectionInputEntity(
+            diameter_projection=Decimal(7),
+            space_between_plants=Decimal(1.5),
+            space_between_rows=Decimal(2)
+        )
+        expected_result = Decimal(
```
[the remark this answers is not in the document]
Comment (geanderson-062) [reply irrigation_design_system_backend/tests/unit/apps/percent_shaded_area/test_reference_percent_shaded_area_service.py]: https://github.com/PoucaAgua/irrigation-design-system-backend/pull/45/commits/88575dc8bdd0da896a11912d64cd784b96e709a3

File: irrigation_design_system_backend/tests/unit/apps/percent_wetted_area/test_reference_percent_wetted_area_service.py (hunk 4 above)
Comment (geanderson-062) [reply irrigation_design_system_backend/tests/unit/apps/percent_wetted_area/test_reference_percent_wetted_area_service.py]: https://github.com/PoucaAgua/irrigation-design-system-backend/pull/45/commits/88575dc8bdd0da896a11912d64cd784b96e709a3

File: irrigation_design_system_backend/apps/percent_shaded_area/Readme.md
Comment (geanderson-062) [reply irrigation_design_system_backend/apps/percent_shaded_area/Readme.md:1]: https://github.com/PoucaAgua/irrigation-design-system-backend/pull/45/commits/d1a9f69a02f7c15ba83e778ff588c4c06d86f568

File: irrigation_design_system_backend/infrastructure/api/v1/responses/percent_wetted_area.py (hunk 2 above)
Comment (geanderson-062) [reply irrigation_design_system_backend/infrastructure/api/v1/responses/percent_wetted_area.py]: https://github.com/PoucaAgua/irrigation-design-system-backend/pull/45/commits/e338d9f99a81df4331d68fa3bf6f99def2bfbe1f

File: irrigation_design_system_backend/core/domain/entity/PWAEntity.py (hunk 1 above)
Comment (geanderson-062) [reply irrigation_design_system_backend/core/domain/entity/PWAEntity.py]: https://github.com/PoucaAgua/irrigation-design-system-backend/pull/45/commits/e06b4787629e07f2bf0fae220a3598a68cebe57a

File: irrigation_design_system_backend/core/domain/entity/PWAEntity.py
Comment (geanderson-062) [reply irrigation_design_system_backend/core/domain/entity/PWAEntity.py:1]: https://github.com/PoucaAgua/irrigation-design-system-backend/pull/45/commits/e06b4787629e07f2bf0fae220a3598a68cebe57a

File: irrigation_design_system_backend/apps/percent_wetted_area/Readme.md (hunk 3 above)
Comment (geanderson-062) [reply irrigation_design_system_backend/apps/percent_wetted_area/Readme.md]: https://github.com/PoucaAgua/irrigation-design-system-backend/pull/45/commits/867c03b6fa7deaf466b9257ea5320f3ffabc5926

**Review by ravellys — APPROVED** [review]
💯 awesome

**Pull request merged**